functional_tests/steps/bundles.py

```diff
@@ -121,3 +128,267 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
 def bundle_inspect_show(context: Context) -> None:
     expect(context.page.get_by_text("Created by")).to_be_visible()
     expect(context.page.get_by_text(context.bundle_creator.get_username())).not_to_be_visible()
+
+
+# Bundles UI Data Setup
+
+
+@given("there is a {user_role} user")
+def a_user_exists_by_role(context: Context, user_role: str) -> None:
+    if not hasattr(context, "users"):
+        context.users = []
+
+    context.users.append({user_role: create_user(user_role)})
+
+
+@given("there are {no_statistical_analysis} Statistical Analysis pages")
+def multiple_statistical_analysis(context: Context, no_statistical_analysis: str) -> None:
+    context.statistical_article_pages = []
+    context.article_series_page = ArticleSeriesPageFactory(title="PSF")
+    if no_statistical_analysis.isdigit():
+        for _ in range(int(no_statistical_analysis)):
+            article = StatisticalArticlePageFactory(parent=context.article_series_page)
+            mark_page_as_ready_to_publish(article, UserFactory())
+            context.statistical_article_pages.append(article)
+
+
+@given("there are {no_release_calendar} release calendar pages")
+def multiple_release_calendar(context: Context, no_release_calendar: str) -> None:
+    context.release_calendar_pages = []
+    if no_release_calendar.isdigit():
+        for release_calendar_index in range(int(no_release_calendar)):
+            nowish = timezone.now() + timedelta(minutes=5 * (release_calendar_index + 1))
+            context.release_calendar_pages.append(
+                ReleaseCalendarPageFactory(
+                    release_date=nowish,
+                    title=f"Release Calendar Page {release_calendar_index}",
+                    status=ReleaseStatus.CONFIRMED,
+                )
+            )
+
+
+@given("there are {no_preview_teams} Preview teams")
+def multiple_preview_teams_create(context: Context, no_preview_teams: str) -> None:
+    context.teams = []
+    if no_preview_teams.isdigit():
+        for preview_team_index in range(int(no_preview_teams)):
+            context.teams.append(
+                Team.objects.create(
+                    identifier=f"preview-team- {preview_team_index}", name=f"Preview_Team_{preview_team_index}"
+                )
+            )
+
+
+@given("the {user_role} is a member of the Preview teams")
+def add_user_to_preview_teams(context: Context, user_role: str) -> None:
+    next((item[user_role]["user"] for item in context.users if item[user_role]), None).teams.add(*context.teams)
+
+
+@given("there are {number_of_bundles} bundles with {bundle_details}")
+def multiple_bundles_create(context: Context, number_of_bundles: str, bundle_details: str) -> None:
+    bundle_dets = {}
+    if json_str_to_dict(bundle_details):
+        bundle_dets = json.loads(bundle_details)
```

Comment:
This looks a bit strange to me, the function you call does `json.loads` and then you catch the exception which returns false. You then attempt json loads again if false is not returned.

I don't think this approach works well. From my point of view this function would never receive invalid json, unless there's a mistake.